polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -340,6 +344,10 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
 
     if (inStickyHeader) {
       this.stickyHeadings[index] = ref;
+
+      const button = ref.querySelector('button');
```

Comment:
Adding focus event listener to only the headings inside of sticky header. When tabbing through the regular table headings, normal keyboard tabbing applies.